### PR TITLE
os/bluestore: Don't pollute old journal when add new device

### PR DESCRIPTION
**Pull request by yanghonggang**
Fixes: https://tracker.ceph.com/issues/44774
Signed-off-by: Yang Honggang <yanghonggang@kuaishou.com>

### CODE REVIEW

File: src/os/bluestore/BlueFS.cc [hunk 1]
```diff
@@ -339,7 +341,9 @@ void BlueFS::_add_block_extent(unsigned id, uint64_t offset, uint64_t length)
   block_all[id].insert(offset, length);
 
   if (id < alloc.size() && alloc[id]) {
-    log_t.op_alloc_add(id, offset, length);
+    if (!skip)
```
Comment (ifed01) [inline src/os/bluestore/BlueFS.cc]: @yanghonggang - I'm not sure I understand what happens after new extent isn't preserved in the log - how  will new device obtain this information after successful attachment?

Comment (yanghonggang) [reply src/os/bluestore/BlueFS.cc]: > @yanghonggang - I'm not sure I understand what happens after new extent isn't preserved in the log - how will new device obtain this information after successful attachment?

new device's extent info will be record to the new log in _rewrite_log_and_layout_sync() -> _compact_log_dump_metadata():

```cpp
  t->op_init();
  for (unsigned bdev = 0; bdev < MAX_BDEV; ++bdev) {
    interval_set<uint64_t>& p = block_all[bdev];
    for (interval_set<uint64_t>::iterator q = p.begin(); q != p.end(); ++q) {
      auto bdev_new = bdev;
      ...
      dout(20) << __func__ << " op_alloc_add " << bdev_new << " 0x"
               << std::hex << q.get_start() << "~" << q.get_len() << std::dec
               << dendl;
      t->op_alloc_add(bdev_new, q.get_start(), q.get_len());
    }
  }
```
And only [this case](https://tracker.ceph.com/issues/44774) need to set skip to true.

Comment (yanghonggang) [reply src/os/bluestore/BlueFS.cc]: ping @ifed01 

how to reproduce this problem: https://tracker.ceph.com/issues/44774?issue_count=65&issue_position=3&next_issue_id=44757&prev_issue_id=44818#How-to-reproduce-this-problem

**Comment by ifed01** [comment]
jenkins retest this please

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2020-04-07_09:41:02-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-04-07_15:13:18-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-04-08_01:39:26-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-04-08_03:18:15-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2020-04-08_04:37:42-rados-wip-kefu-testing-2020-04-07-1603-distro-basic-smithi/

**Comment by tchaikov** [comment]
@yanghonggang @ifed01 do we need to backport this change? as per the description of the tracker ticket, it sounds that this issue would cause data unavailability issues.

**Comment by yanghonggang** [comment]
> @yanghonggang @ifed01 do we need to backport this change? as per the description of the tracker ticket, it sounds that this issue would cause data unavailability issues.

I think it should be backported @tchaikov

**Pull request merged**